pkg/envoy/ads/stream.go

```diff
@@ -116,6 +116,8 @@ func (s *Server) StreamAggregatedResources(server xds_discovery.AggregatedDiscov
 
 			<-s.workqueues.AddJob(newJob(typesRequest, &discoveryRequest))
 
+			metricsstore.DefaultMetricsStore.ProxyXDSRequestCount.WithLabelValues(certCommonName.String(), discoveryRequest.TypeUrl).Inc()
```

Comment:
Ideally, this counter should be incremented for every request, not just the requests that are responded to. So it would make sense to move it above L110. Keeping it here only counts the number of requests that are going to be responded to, which should ideally be similar to the number of responses sent.

This metric would be valuable to count all requests from Envoy, even those that are not responded to, as it would help determine whether the controller is ignoring requests when this counter keeps incrementing but there are no responses from the contoller.